storage/option.go

```diff
@@ -108,3 +111,59 @@ func WithDisabledClientMetrics() option.ClientOption {
 func (w *withDisabledClientMetrics) ApplyStorageOpt(c *storageConfig) {
 	c.disableClientMetrics = w.disabledClientMetrics
 }
+
+// WithDynamicReadReqStallTimeout is an option that may be passed to [NewClient].
+// It enables the client to retry the stalled read request, happens as part of
+// storage.Reader creation. As the name suggest, timeout is adjusted dynamically
+// based on past observed read-req latencies.
+//
+// This is only supported for the read operation and that too for http(XML) client.
+// Grpc read-operation will be supported soon.
+//
+// Here, the input parameter decides the value of dynamic-timeout.
+// targetPercentile is the desired percentile of the observed latencies.
+// increaseRate determines the rate, timeout is adjusted. High means slow increase in
+// timeout and low means rapid increase.
+// initialTimeout decides the initial timeout.
+// minTimeout, maxTimeout is the lower & upper bound of the timeout.
+//
+// TODO (raj-prince): To keep separate dynamicDelay instance for different BucketHandle.
+// Currently, dynamicTimeout is kept at the client and hence shared across all the
+// BucketHandle, which is not not the ideal state. As latency depends on location of VM
+// and Bucket, and read latency of different buckets may lie in different range.
+// Hence hence having a separate dynamicTimeout instance at BucketHandle level will
+// be better
+func WithDynamicReadReqStallTimeout(
+	targetPercentile float64,
+	increaseRate float64,
+	initialTimeout time.Duration,
+	minTimeout time.Duration,
+	maxTimeout time.Duration) option.ClientOption {
```

Comment:
I think we could reduce the surface area here. Only targetPercentile and minTimeout are really crucial. increaseRate can be set internally (not sure people would do this level of tweaking), initialTimeout could be equal to minTimeout, and maxTimeout should probably be infinite (or very high) for safety reasons.

Reply:
Let's also change the args to a config struct so that we can add stuff in the future if it does end up being needed.

Reply:
We can hardcore the value of maxTimeout. 

For increaseRate - I am not very sure if we should hardcore the value of increaseRate. Although less likely but we might need to change the value of increaseRate based on testing.  Is it possible to expose these as part of experimental flag and later remove it if not required.

For initialTimeout - I was thinking to keep the initial value slightly higher than minTimeout. This will give some time to the dynamicDelay algorithm to converge.

>> Let's also change the args to a config struct so that we can add stuff in the future if it does end up being needed. 
- Yeah, make sense.

Reply:
As discussed, we will keep large maxTimeout. increaseRate and initialTimeout (not discussed, but still feel like we might need to change this) will be exposed via environment variable.